### PR TITLE
metadata refactor

### PR DESCRIPTION
**Pull request by murali-shris**

### CODE REVIEW

File: at_secondary/at_persistence_secondary_server/lib/src/keystore/hive_keystore.dart [hunk 1]
```diff
@@ -44,25 +44,18 @@ class HiveKeystore implements SecondaryKeyStore<String, AtData, AtMetaData> {
...
     var commitOp = CommitOp.UPDATE;
     // Verifies if any of the args are not null
     var isMetadataNotNull = ObjectsUtil.isAnyNotNull(
-        a1: time_to_live,
-        a2: time_to_born,
-        a3: time_to_refresh,
-        a4: isCascade,
-        a5: isBinary,
-        a6: isEncrypted);
+        a1: metadata.ttl,
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/keystore/hive_keystore.dart:53]: I hope metadata itself is not null

**Review by VJag — APPROVED** [review]
Great.

File: at_persistence/at_persistence_spec/lib/src/keystore/keystore.dart [hunk 2]
```diff
@@ -18,14 +20,7 @@ abstract class WritableKeystore<K, V> implements Keystore<K, V> {
...
   /// Throws a [DataStoreException] if the the operation fails due to some issue with the data store.
-  Future<dynamic> put(K key, V value,
-      {int time_to_live,
-      int time_to_born,
-      int time_to_refresh,
-      bool isCascade,
-      bool isBinary,
-      bool isEncrypted,
-      String dataSignature});
+  Future<dynamic> put(K key, V value, {Metadata metadata});
```
Comment (murali-shris) [inline at_persistence/at_persistence_spec/lib/src/keystore/keystore.dart:23]: please modify dart doc.

**Comment by murali-shris** [comment]
@naresh0689 please add functional test to check shared key status update

**Pull request closed without merging**